src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Xaml/Replacements/TypeListConverter.cs

```diff
@@ -4,50 +4,26 @@
 
 using System.ComponentModel;
 using System.Globalization;
-using XAML3 = System.Windows.Markup;
 
 namespace System.Xaml.Replacements
 {
-    // Not sure if this type converter is used at all.
-    // we need to either make this a useful type converter or remove the code.
-    
     /// <summary>
     /// TypeConverter for System.Type[]
     /// </summary>
     internal class TypeListConverter : TypeConverter
     {
-        private static readonly TypeTypeConverter typeTypeConverter = new TypeTypeConverter();
         public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
-        {
-            return sourceType == typeof(string);
-        }
+            => sourceType == typeof(string);
 
         public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
         {
             string typeList = (string)value;
-            if (null != context)
+            if (context != null)
             {
-                // Consider HashMap(int, int), HashMap(int, int)
-                string[] tl = StringHelpers.SplitTypeList(typeList);
-                Type[] types = new Type[tl.Length];
-                for (int i = 0; i < tl.Length; i++)
-                {
-                    types[i] = (Type)typeTypeConverter.ConvertFrom(context, TypeConverterHelper.InvariantEnglishUS, tl[i]);
-                }
-                return types;
+                throw new NullReferenceException();
```

Comment:
I think this code may be controversial, why throw a null exception when the variable is not null?

Reply:
Yeah - basically this always used to throw `NullReferenceException`, because `StringHelpers.SplitTypeList` always returns null and then `tl.Length` would throw `NullReferenceException`.

Welcome comments here.

Reply:
Hey @hughbe ,
We are considering merging this PR. I'd like to offer a suggestion: rather than removing the code and returning ```null``` , we could set it up like this:
```public static string[] SplitTypeList(string typeList) => Array.Empty<string>(); ```

With this approach, ```StringHelpers.SplitTypeList``` will return an empty string, allowing the code to proceed. This setup also leaves the door open for us to potentially utilize this feature in the future.

Thank you for your consideration.